security/pkg/server/ca/server.go

```diff
@@ -35,11 +36,10 @@ var serverCaLog = log.RegisterScope("serverca", "Citadel server log", 0)
 
 // CertificateAuthority contains methods to be supported by a CA.
 type CertificateAuthority interface {
-	// Sign generates a certificate for a workload or CA, from the given CSR and TTL.
-	// TODO(myidpt): simplify this interface and pass a struct with cert field values instead.
-	Sign(csrPEM []byte, subjectIDs []string, ttl time.Duration, forCA bool) ([]byte, error)
+	// Sign generates a certificate for a workload or CA, from the given CSR and cert opts.
+	Sign(csrPEM []byte, opt *ca.CertOpts) ([]byte, error)
 	// SignWithCertChain is similar to Sign but returns the leaf cert and the entire cert chain.
-	SignWithCertChain(csrPEM []byte, subjectIDs []string, ttl time.Duration, forCA bool) ([]byte, error)
+	SignWithCertChain(csrPEM []byte, opt *ca.CertOpts) ([]byte, error)
```

Comment:
I am not sure if there are benefits to making it a pointer. 

In terms of performance, its likely worse since we need to heap allocate instead of stack allocate and the struct is tiny

In terms of readability, it is more complex since the function is now allowed to mutate the struct

Reply:
Thanks for the comments! I was also concerned about the benefits brought by passing it as a pointer, while making the change.
Personally I don't think there are benefits to making it a pointer at this moment (though there might be in the future if one extends it w/ much more fields... or need mutability within Sign* methods).
Changed to pass by value for now.